src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -192,12 +194,14 @@ internal override bool IsExpressionBodied
             MethodKind methodKind,
             bool isAutoPropertyAccessor,
             DiagnosticBag diagnostics) :
-            base(containingType, syntax.GetReference(), syntax.Body?.GetReference(), location)
+            base(containingType, syntax.GetReference(), syntax.Body?.GetReference() ?? syntax.ExpressionBody?.GetReference(), location)
         {
             _property = property;
             _explicitInterfaceImplementations = explicitInterfaceImplementations;
             _name = name;
             _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            Debug.Assert(_property.IsExpressionBodied == false, "Cannot have accessors in expression bodied lightweight properties");
+            _isExpressionBodied = (syntax.Body == null && syntax.ExpressionBody != null);
```

Comment:
>syntax.Body == null [](start = 35, length = 19)

Consider factoring `syntax.Body == null`. Then you don't need the `bodyOpt` local below.